### PR TITLE
runtimetest: add root filesystem validation

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

**Comment by liangchenye** [comment]
Hi @Mashimiao I just submitted a PR https://github.com/opencontainers/ocitools/pull/81 which does the same thing, but by using mount information.

I think it might be wrong to verify the `readonly property` of the root file system by using os.stat.
if you `mount yourPartition -o /mnt/readonly-point` and use os.stat, it might still have write permission. But when you touch `a` file in /tmp/readonly-point, it will tell:
```
touch: cannot touch 'a': Read-only file system
```

**Comment by wking** [comment]
On Thu, May 26, 2016 at 03:43:54AM -0700, 梁辰晔 (Liang Chenye) wrote:
> if you `mount yourPartition -o /mnt/readonly-point` and use os.stat,
> it might still have write permission. But when you touch `a` file in
> /tmp/readonly-point…

Ah, good point.  We should definitely be testing read-only by
attempting writes, because the per-file POSIX-mode permissions are
distinct from the per-mount filesystem options.

**Comment by Mashimiao** [comment]
@liangchenye  @wking   re-implemented with pre-write

**Comment by wking** [comment]
On Thu, May 26, 2016 at 09:47:37PM -0700, Ma Shimiao wrote:
> @liangchenye  @wking   re-implemented with pre-write

I'll leave it to you and @liangchenye to decide whether this rerolled
version takes precedence over #81 ;).

**Comment by liangchenye** [comment]
@wking  Ah! I'll rollback my second commit in #81.

File: cmd/runtimetest/main.go [hunk 1]
```diff
@@ -191,13 +191,38 @@ func validateSysctls(spec *rspec.Spec) error {
...
 }
 
+func testWriteAccess(path string) error {
+	tmpfile, err := ioutil.TempFile(path, "Test")
+	if err != nil {
+		return err
+	}
+
+	tmpfile.Close()
+	os.RemoveAll(filepath.Join(path,tmpfile.Name()))
```
Comment (mrunalp) [inline cmd/runtimetest/main.go]: Probably needs gofmt run.

Comment (Mashimiao) [inline cmd/runtimetest/main.go]: @mrunalp    Ah, sorry. fixed.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**